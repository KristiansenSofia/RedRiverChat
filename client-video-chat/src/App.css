.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  height: 100% !important;
  width: 100% !important;
}

.NameForm {
  margin: auto;
}

.UserList {
  width: 25%;
  margin-left: 5px;
  text-align: left;
  height: 100%;
  border-right: 1px solid #cbced0;
}

.VideoChat-Main {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: row;
}

.VideoChat {
  width: 75%;
  height: 100%;
}

.UserList-Ul {
  display: flex;
  flex-direction: column;
  list-style: none;
  text-align: left;
  padding: 5px;
}

.UserList-li {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.CallButtons {
}

.Button {
  height: 30px;
  width: 30px;
  border-radius: 50%;
  background-color: #cbced0;
  margin: 5px;
}

.Button:hover {
  background-color: #878a8c;
}

.Button:focus {
  outline: none;
}