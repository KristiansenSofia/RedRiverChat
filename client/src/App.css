.App {
  text-align: center;
}

.AppBar {
  flex-flow: 1;
}