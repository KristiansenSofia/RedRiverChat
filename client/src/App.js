import React, { Component } from 'react'
import { BrowserRouter as Router, Link, Route } from 'react-router-dom'

// Import NPM-modules
import { MuiThemeProvider } from 'material-ui/styles'
import AppBar from 'material-ui/AppBar'
import Toolbar from 'material-ui/Toolbar'
import Typography from 'material-ui/Typography'
import IconButton from 'material-ui/IconButton'
import MenuIcon from '@material-ui/icons/Menu'
import Drawer from 'material-ui/Drawer'
import List, { ListItem, ListItemIcon, ListItemText } from 'material-ui/List'
import Divider from 'material-ui/Divider'
import Snackbar from 'material-ui/Snackbar'
import { CircularProgress } from 'material-ui/Progress'
import HttpsRedirect from 'react-https-redirect'

// Import icons for the drawer-menu.
import ChatIcon from '@material-ui/icons/ChatBubble'
import PersonIcon from '@material-ui/icons/People'
import SettingsIcon from '@material-ui/icons/Settings'
import LoginIcon from '@material-ui/icons/Person'
import LogoutIcon from '@material-ui/icons/Cancel'

// Import styles. appStyles for all imported components with a style attribute and CSS-file for classNames and id.
import './styles/Styles.css'
import {theme} from './styles/Styles'
import AppStyles from './styles/AppStyles'

// Import pages to use with React Router for navigation.
import ChatList from './component/chat/ChatList'
import FriendsList from './component/friends/FriendsList'
import Settings from './component/settings/Settings'
import Login from './component/authentication/Login'
import Register from './component/authentication/Register'
import NewPassword from './component/authentication/NewPassword'
import UserAccount from './component/account/UserAccount'
import FriendRequests from './component/friends/FriendRequests'

import {verifyJWT} from './utils/ApiRequests'
import {createSignalR, initChat} from './utils/SignalR'

/**
 *  Starting point of the application
 *
 *  @author Jimmy
 */

class App extends Component {
  constructor (props) {
    super(props)

    this.state = {
      menu: false,
      snackBar: false,
      snackBarMessage: '',
      isSignedIn: false,
      userRole: 'User',
      loaded: false,
      signalRConnection: ''
    }
    this.openSnackBar = this.openSnackBar.bind(this)
    this.userLogout = this.userLogout.bind(this)
    this.userLogin = this.userLogin.bind(this)
  }

  /**
     *  Logout. Delete token in local storage and change state.isSignedIn to false.
     *
     *  @author Jimmy
     */

  userLogout () {
    this.setState({
      isSignedIn: false
    })
    localStorage.removeItem('token')
    localStorage.removeItem('userInfo')
  }

  /**
   *  Login. Set state with user details
   *
   *  @author Jimmy
   */

  userLogin (token) {
    this.verifyToken(token)
  }

    /**
     *  Open bottom-bar and display message. Closes after 3 seconds.
     *
     *  @author Jimmy
     */

    openSnackBar = (message) => {
      this.setState({
        snackBar: true,
        snackBarMessage: message
      })

      setTimeout(() => {
        this.closeSnackBar()
      }, 3000)
    };

    /**
     *  Close bottom-bar and delete message.
     *
     *  @author Jimmy
     */

    closeSnackBar = () => {
      this.setState({
        snackBar: false,
        snackBarMessage: ''
      })
    };

    /**
     *  Render all links in drawer-menu.
     *
     *  @author Jimmy
     */

    renderMenu = () => {
      return (
        <div style={AppStyles.menu}>
          <List>
            <ListItem
              button
              component={Link}
              to='/'
            >
              <ListItemIcon>
                <LoginIcon />
              </ListItemIcon>
              <ListItemText primary='Start' />
            </ListItem>
            <ListItem
              button
              component={Link}
              to='/chats'
            >
              <ListItemIcon>
                <ChatIcon />
              </ListItemIcon>
              <ListItemText primary='Chat' />
            </ListItem>
            <ListItem
              button
              component={Link}
              to='/friends'
            >
              <ListItemIcon>
                <PersonIcon />
              </ListItemIcon>
              <ListItemText primary='Vänner' />
            </ListItem>
            <ListItem
              button
              component={Link}
              to='/settings'
            >
              <ListItemIcon>
                <SettingsIcon />
              </ListItemIcon>
              <ListItemText primary='Inställningar' />
            </ListItem>
          </List>
          <Divider />
          <ListItem
            button
            component={Link}
            to='/'
            onClick={this.userLogout}
          >
            <ListItemIcon>
              <LogoutIcon />
            </ListItemIcon>
            <ListItemText primary='Logga ut' />
          </ListItem>
        </div>
      )
    };

    /**
     *  Open drawer-menu.
     *
     *  @author Jimmy
     */

    toggleMenu = (open) => () => {
      this.setState({
        menu: open
      })
    };

  /**
   *  Verify token and set state according to response from server.
   *  If token is valid, a chat-socket is initialized.
   *
   *  @author Jimmy
   */

    verifyToken (token) {
      return verifyJWT(token)
        .then((response) => {

          initChat(token)

          this.setState({
            token: token,
            isSignedIn: true,
            userInfo: response.data,
            loaded: true
          })
        }).catch(() => {
          this.setState({
            isSignedIn: false,
            loaded: true
          })
        })
    }

    /**
     *  Check if valid token in local storage before component mounts.
     *
     *  @author Jimmy
     */

    componentWillMount () {
      if (localStorage.getItem('token')) {
        let token = JSON.parse(localStorage.getItem('token'))
        this.verifyToken(token)
      } else {
        this.setState({
          isSignedIn: false,
          loaded: true
        })
      }
    }

    /**
     *  Check if valid token in local storage before component updates.
     *
     *  @author Jimmy
     */

    componentWillUpdate () {
      if (localStorage.getItem('token')) {
        let token = JSON.parse(localStorage.getItem('token'))

        if (this.state.token) {
          if (token !== this.state.token) {
            this.verifyToken(token)
          }
        }
      }
    }

    render () {
      return (
        <HttpsRedirect>
          <Router>
            <MuiThemeProvider theme={theme} >
              {this.state.loaded ? (
                <div className='App'>
                  <div className='App'>
                    <AppBar
                      position='sticky'
                      style={AppStyles.root}
                    >
                      <Toolbar>
                        <Typography
                          variant='title'
                          color='inherit'
                          style={AppStyles.flex}
                        />
                        {this.state.isSignedIn ? (
                          <IconButton color='inherit' aria-label='Menu' style={AppStyles.menuButton} onClick={this.toggleMenu(true)}>
                            <MenuIcon />
                          </IconButton>
                        ) : (
                          <p />
                        )}
                      </Toolbar>
                    </AppBar>
                    <div className='App-Body'>
                      <Route path='/' exact component={() => <UserAccount state={this.state} />} />
                      <Route path='/chats' component={() => <ChatList state={this.state} />} openSnackBar={this.openSnackBar} />
                      <Route path='/friends' component={() => <FriendsList state={this.state} openSnackBar={this.openSnackBar} />} />
                      <Route path='/settings' component={() => <Settings state={this.state} />} />
                      <Route path='/login' component={() => <Login state={this.state} openSnackBar={this.openSnackBar} userLogin={this.userLogin} />} />
                      <Route path='/register' component={() => <Register state={this.state} openSnackBar={this.openSnackBar} />} />
                      <Route path='/password' component={() => <NewPassword state={this.state} />} />
                      <Route path='/friendrequests' component={() => <FriendRequests state={this.state} openSnackBar={this.openSnackBar} />} />
                    </div>
                    <Snackbar
                      open={this.state.snackBar}
                      onClose={this.closeSnackBar}
                      SnackbarContentProps={{
                        'aria-describedby': 'message-id'
                      }}
                      message={<span id='message-id'>{this.state.snackBarMessage}</span>}
                    />
                    <Drawer anchor='right' open={this.state.menu} onClose={this.toggleMenu(false)}>
                      <div
                        tabIndex={0}
                        role='button'
                        onClick={this.toggleMenu(false)}
                        onKeyDown={this.toggleMenu(false)}
                      >
                        {this.renderMenu()}
                      </div>
                    </Drawer>
                  </div>
                </div>
              ) : (
                <div className='AppLoadingDiv'>
                  <CircularProgress style={AppStyles.loading} />
                </div>
              )}
            </MuiThemeProvider>
          </Router>
        </HttpsRedirect>
      )
    }
}

export default App
