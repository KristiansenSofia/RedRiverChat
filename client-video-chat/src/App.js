import React, { Component } from 'react';
import './App.css';

import VideoChat from './components/VideoChat';
import NameForm from './components/NameForm';

class App extends Component {
    constructor(props) {
        super(props);

        this.state = {
            name: '',
        };
        this.handleNameSubmit = this.handleNameSubmit.bind(this)
    }

    handleNameSubmit(value) {

      this.setState({name: value,});
      sessionStorage.setItem('name', value);
    }

    componentWillMount() {

      if (sessionStorage.getItem('name')) {

          this.setState({name: sessionStorage.getItem('name'),});
      }
    }

  render() {
    return (
      <div className="App">
          {this.state.name ? (
              <div className="App">
                  <VideoChat/>
              </div>
          ) : (
              <NameForm handleNameSubmit={this.handleNameSubmit}/>
          )}
      </div>
    );
  }
}

export default App;
