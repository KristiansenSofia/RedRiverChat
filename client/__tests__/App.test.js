import React from 'react';
import App from '../src/App';
import renderer from 'react-test-renderer';

test('App renders without crashing', () => {
  const tree = renderer.create(<App />);
  expect(tree.toJSON()).toMatchSnapshot();
});
