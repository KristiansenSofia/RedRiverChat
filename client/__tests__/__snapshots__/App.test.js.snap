// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div>
  <div
    className="App"
  >
    <header
      className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionSticky-4 MuiAppBar-colorPrimary-7"
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <div
        className="MuiToolbar-root-36 MuiToolbar-gutters-37"
      >
        <h2
          className="MuiTypography-root-38 MuiTypography-title-44 MuiTypography-colorInherit-57"
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <p>
           
        </p>
      </div>
    </header>
    <div
      className="Body"
    />
  </div>
</div>
`;
